src/connect4/core.rs

```diff
@@ -366,26 +383,80 @@ impl Board {
     pub fn insert(&mut self, position: i32, team: i32, color: MyColor) -> bool {
         self.columns[position as usize].insert(team, color)
     }
+    
+    pub fn reset(&mut self) {
+        for column in &mut self.columns {
+            column.reset();
+        }
+    }
+}
 
+pub struct TurnIndicator {
+    team: i32,
 }
 
+impl TurnIndicator {
+    pub fn new() -> Self {
+        TurnIndicator {
+            team: 0,
+        }
+    }
+
+    fn draw(&self, ctx: &mut Context) -> GameResult<()> {
+        let mut text: graphics::Text;
+        if self.team == 0 {
+            text = graphics::Text::new(("Paused", graphics::Font::default(), TURN_INDICATOR_FONT_SIZE as f32));
+        } else {
+            text = graphics::Text::new((format!("Player {}'s turn", self.team), graphics::Font::default(), TURN_INDICATOR_FONT_SIZE as f32));
+        }
+        let dim = &text.dimensions(ctx);
+        let pos = Point2{
+            x: TURN_INDICATOR_POS_OFFSET.0 as f32 - (dim.0 as f32 / 2.0) as f32, 
+            y: TURN_INDICATOR_POS_OFFSET.1 as f32
+        };
+
+        let textbox = graphics::Mesh::new_rectangle(
+            ctx, 
+            graphics::DrawMode::fill(),             
+            graphics::Rect {
+                x: pos.x, 
+                y: pos.y, 
+                w: dim.0 as f32 + TURN_INDICATOR_BOX_SIZE_OFFSET.0 as f32,
+                h: dim.1 as f32 + TURN_INDICATOR_BOX_SIZE_OFFSET.1 as f32,
+            },
+            graphics::Color::from_rgba(205,133,63,255),
+        )?;
+        graphics::draw(ctx, &textbox, (Point2 {x: 0.0, y: 0.0},))?;
+        graphics::draw(ctx, &text, (Point2 {x: pos.x + TURN_INDICATOR_BOX_SIZE_OFFSET.0 as f32 / 2.0, y: pos.y + TURN_INDICATOR_BOX_SIZE_OFFSET.1 as f32 / 2.0},))?;
```

Comment:
Additional addition to the `pos.x` and `pos.y` so the text is centered in the `textbox`.

Reply:
Nice! Cool little addition